src/Surfnet/StepupMiddleware/ApiBundle/Identity/Projector/AuditLogProjector.php

```diff
@@ -91,6 +99,11 @@ private function applyAuditableEvent(AuditableEvent $event, DomainMessage $domai
             $entry->actorId         = $metadata['actorId'];
             $entry->actorCommonName = $actor->commonName;
         }
+        if ($event instanceof SecondFactorVettedEvent ||
```

Comment:
Why isn't this coming from the metadata of the event? It seems to me that the responsibility of the metadata is to do this.

Reply:
Please elaborate

Reply:
All loggable information currently comes from the metadata of the event. So by just checking if the metadata exists you prevent coupling with concrete Event implementations in the AuditlogProjector. You could also get rid of the VettingType::auditLog method because this auditlog logic doesn't seem to belong in this VO.

Reply:
Good point, I completely ovetlooked this. The AuditableEvent can be used neatly for this intent. For now I will leave the retrieval of the audit log entry on the VO. I feel it is the responsibility of the vetting type to determine how its audit log entry is composed. Making this the responsibility of the projector is also not a pure responsibility of the projector in my opinion.